### ISSUE TITLE
Validate schemas with a top level list

### ISSUE THREAD

**Issue by wilrodriguez** [ryan95f/yamlator#46]
At the moment, it's not really possible to validate a schema where the top level datastructure is a list. This is pretty common in a lot of places and probably is something for future enhancement. You might have a datastructure like so:

```yaml
- foo:
   bar: true
   baz: false
- qux:
  bar: false
  baz: true
```

And it would be nice to be able to validate it with something like a special rule key called something like `!!rootlist` and be able to define a data structure like so:

```
strict ruleset Foobar {
    bar bool
    baz bool
}

schema {
    !!rootlist Foobar
}
```

Alternatively, it could just be a different kind of schema, something like this:

```
listschema {
    Foobar
}
```

**Comment by ryan95f** [comment]
This is a limitation that I have known about for a while. I have had some ideas around this since it would also need to support objects / maps, strings and ints which are all valid YAML without a key.  I do like the use of `!!root` to represent a keyless field. Though there is that one chance someone has a valid key of `!!root`, which in that case I would need to check the data structure to figure out if this is a field with a key / without.

**Comment by wilrodriguez** [comment]
hmmm maybe maybe it even more specific like `!!yamlatorlist`, I highly doubt anyone would have that, and if they did I think at that point we can safely call that user error.

As far as supporting the other things that don't have a key, I think that would probably be lower priority since those aren't used as commonly on their own, certainly not in the way that it would be a priority to be able to validate them.

**Comment by ryan95f** [comment]
The limitation with only supporting objects has been resolved as part of `0.3.1` with the inclusion of top level validation for any data type. To do this, in your schema block define a single rule with the name `!!yamlator` then specify the type, for example:

```text
ruleset Foo{
    ...
}

schema {
    !!yamlator list(Foo)
}
```

This will allow for lists to be defined at the very top level or any other types :)

**Comment by wilrodriguez** [comment]
Nice! I think will definitely resolve it!
